h2o-r/tests/runitUtils/utilsR.R

```diff
@@ -931,17 +931,22 @@ random_NN <- function(actFunc, max_layers, max_node_number) {
 # Parameters:  frame1, frame2: H2O frames to be compared.
 #              tolerance: tolerance of comparison
 #----------------------------------------------------------------------
-compareFrames <- function(frame1, frame2, prob=0.5, tolerance=1e-6) {
+compareFrames <- function(frame1, frame2, prob=0.5, tolerance=1e-6, enum2String=FALSE) {
   expect_true(nrow(frame1) == nrow(frame2) && ncol(frame1) == ncol(frame2), info="frame1 and frame2 are different in size.")
   for (colInd in c(1:ncol(frame1))) {
 
     notNumericCols = !(h2o.isnumeric(frame1[,colInd]) && h2o.isnumeric(frame2[,colInd]))
     if (notNumericCols) {
-      temp1 = frame1[,colInd]
-      temp2 = frame2[,colInd]
+      if (enum2String) {
+        temp1 = as.data.frame(h2o.ascharacter(frame1[,colInd]))
+        temp2 = as.data.frame(h2o.ascharacter(frame2[,colInd]))       
+      } else {
+        temp1 = as.data.frame(frame1[,colInd])
+        temp2 = as.data.frame(frame2[,colInd])
+      }
     } else { 
-      temp1=as.numeric(frame1[,colInd])
-      temp2=as.numeric(frame2[,colInd])
+      temp1=as.data.frame(frame1[,colInd])
```

Comment:
why do you need the as.data.frame everywhere?

Reply:
I want to grab all data in one shot from H2O to R.  By using as.data.frame, R will grab one column of H2O Data.  If I just do frame1[, colInd] and then later on when I iterate over each temp1[ind], H2O will give R one element at a time and this can be slow especially if the dataset is large.

Reply:
I change the code so that as.data.frame is called only once.

Reply:
in python tests we have two different functions for that compare_frames and compare_frames_local

https://github.com/h2oai/h2o-3/blob/b9e241e19f64e876f57ecdfa9fd6cd6f865dfebd/h2o-py/tests/pyunit_utils/utilsPY.py#L3468

does it make sense to have the same dual approach available in R tests? I am not sure why you would ever use the non-local version since its so much slower, but perhaps there is one?

Reply:
Nope, the nonlocal version was written first.  Then I added the local version.